constantine/math_compiler/experimental/nim_to_gpu.nim

```diff
@@ -1127,14 +1130,18 @@ proc toGpuAst*(ctx: var GpuContext, node: NimNode): GpuAst =
     result = GpuAst(kind: gpuLit)
     result.lValue = $node.intVal
     result.lType = initGpuType(gtInt32)
-  of nnkUInt32Lit:
+  of nnkUIntLit, nnkUint64Lit:
     result = GpuAst(kind: gpuLit)
     result.lValue = $node.intVal
-    result.lType = initGpuType(gtUInt32)
-  of nnkUIntLit:
+    result.lType = initGpuType(gtUInt64) ## XXX: base on target platform!
+  of nnkUInt16Lit:
     result = GpuAst(kind: gpuLit)
     result.lValue = $node.intVal
-    result.lType = initGpuType(gtUInt64) ## XXX: base on target platform!
+    result.lType = initGpuType(gtUInt16) ## XXX: base on target platform!
+  of nnkUInt32Lit:
+    result = GpuAst(kind: gpuLit)
+    result.lValue = $node.intVal
+    result.lType = initGpuType(gtUInt32) ## XXX: base on target platform!
   of nnkFloat64Lit, nnkFloatLit:
     result = GpuAst(kind: gpuLit)
     result.lValue = $node.floatVal & "f"
```

Comment:
iirc, f is float32, and default float in C/Metal/Cuda is float32 not float64 iirc